Automattic-Tracks-iOS/Model/Core Data/TracksContextManager.h

```diff
@@ -3,8 +3,22 @@
 
 @interface TracksContextManager : NSObject
 
-@property (nonatomic, strong) NSManagedObjectContext *managedObjectContext;
-@property (nonatomic, strong) NSManagedObjectModel *managedObjectModel;
-@property (nonatomic, strong) NSPersistentStoreCoordinator *persistentStoreCoordinator;
+@property (nonnull, nonatomic, strong) NSManagedObjectContext *managedObjectContext;
+@property (nonnull, nonatomic, strong) NSManagedObjectModel *managedObjectModel;
+@property (nonnull, nonatomic, strong) NSPersistentStoreCoordinator *persistentStoreCoordinator;
+
+/// In version 0.5.1 and earlier, the CoreData store was always saved in the Documents folder.
+///
+/// This approach is fine for sandboxed applications distributed via the Mac App Store, but would
+/// result in the user being asked for access to the `~/Documents` folder in non-sandboxed apps.
+///
+/// To avoid this bad UX – which also blocks tests in CI with the same alert – when the app is
+/// sandboxed, we can save the store in `~/Application Support`.
+///
+/// This designated initializer allows consumers to tell `TracksContextManager` the mode in which
+/// the app is running and where to save the store.
+///
+/// The `super` `init` method defaults to `false` for backward compatibility.
+- (nonnull instancetype)initWithSandboxedMode:(BOOL)sandboxed NS_DESIGNATED_INITIALIZER;
```

Comment:
I wonder whether the parameter should be an enum instead:

```swift
_ = TracksContextManager(storeLocation: .applicationSupport)
// or
_ = TracksContextManager(storeLocation: .documents)
```

I'm on the fence and might sit overnight. Regardless, because the changes doesn't affect existent consumers thanks to the `init` override below, I think it's okay to merge this as it is and then I can follow up with a refactor if we think it's worth it.

Reply:
I'm ok with either approach. I like how pretty the enum is, but I like how much work @jleandroperez's isn't 😄